query-engine/core/src/interactive_transactions/actor_manager.rs

```diff
@@ -91,19 +101,33 @@ impl TransactionActorManager {
         operations: Vec<Operation>,
         trace_id: Option<String>,
     ) -> crate::Result<Vec<crate::Result<ResponseData>>> {
+        self.handle_closed_tx(tx_id, "batch query").await?;
         let client = self.get_client(tx_id).await?;
         client.batch_execute(operations, trace_id).await
     }
 
     pub async fn commit_tx(&self, tx_id: &TxId) -> crate::Result<()> {
+        self.handle_closed_tx(tx_id, "commit").await?;
         let client = self.get_client(tx_id).await?;
         client.commit().await?;
         Ok(())
     }
 
     pub async fn rollback_tx(&self, tx_id: &TxId) -> crate::Result<()> {
+        self.handle_closed_tx(tx_id, "rollback").await?;
         let client = self.get_client(tx_id).await?;
         client.rollback().await?;
         Ok(())
     }
+
+    async fn handle_closed_tx(&self, tx_id: &TxId, operation: &str) -> crate::Result<()> {
```

Comment:
Sorry a very minor nit. But I think this would be better named `check_if_tx_closed` or something like that.